zerver/tests/test_rate_limiter.py

```diff
@@ -165,13 +166,45 @@ def test_block_access(self) -> None:
         obj.block_access(1)
         self.make_request(obj, expect_ratelimited=True, verify_api_calls_left=False)
 
-class RateLimitedUserTest(ZulipTestCase):
+class TornadoInMemoryRateLimiterBackendTest(RateLimiterBackendBase):
+    __unittest_skip__ = False
+    backend = TornadoInMemoryRateLimiterBackend
+
+    def api_calls_left_from_history(self, history: List[float], max_window: int,
+                                    max_calls: int, now: float) -> Tuple[int, float]:
+        reset_time = 0.0
+        for timestamp in history:
+            reset_time = max(reset_time, timestamp) + (max_window / max_calls)
+
+        calls_left = (now + max_window - reset_time) * max_calls // max_window
+        calls_left = int(calls_left)  # mypy considers the above expression to be float for some reason
```

Comment:
Ah, I see that due to the timestamp being `float`, the expression actually is a `float` too in the sense of `<integer>.0`, so I understand why mypy behaved this way. On the other hand, wouldn't it make sense to expect passing it as `int` type to be considered valid by mypy - since it is actually an integer in the "mathematical" sense?